### DISCUSSION TITLE
kvlistValue not converted to a JSON string

### DISCUSSION THREAD

**Discussion by rcollette** [microsoft/aspire#5872] [Q&A] [answered]
I'm using the Serilog OpenTelemetrySink.  My structured logs contain kvlistValue objects for my structured logs.
![image](https://github.com/user-attachments/assets/01ed30ad-daf8-453f-a8df-c51974c04aae)

I see in the codebase that there is a converter to turn these into JSON strings, but I am wondering why it is not happening in this case?  I wanted to check if this intended, or some misconfiguration on my part, before I file a bug.

**Update**
If I enable OLTP export for logging with the OpenTelemetry SDK/package, the LogContext appears as JSON as I would expect, like: 
```
{"login":"someone@some.org"}
```

This makes it seem like it could be a bug in the OpenTelemetrySink, maybe an extra set of quotes or something in what they are sending, except that I would expect an issue to have been filed about it by now. The same could be said here as well, which is why I am hesitant to file a bug... it could just be me.

```
        services.AddLogging(
            loggingBuilder =>
            {
                LoggerConfiguration loggingConfiguration = new LoggerConfiguration()
                    .ReadFrom.Configuration(configuration);

                string? otelExporterEndpoint = Configuration["OTEL_EXPORTER_OTLP_ENDPOINT"];
                if (otelExporterEndpoint != null)
                {
                    loggingConfiguration.WriteTo.OpenTelemetry(
                        opt =>
                        {
                            opt.Endpoint = otelExporterEndpoint;
                            opt.Protocol = OtlpProtocol.Grpc;
                            opt.ResourceAttributes = _otelResourceAttributes;
                            opt.OnBeginSuppressInstrumentation =
                                SuppressInstrumentationScope.Begin;
                            opt.IncludedData = IncludedData.TemplateBody | IncludedData.SpanIdField |
                                               IncludedData.TraceIdField | IncludedData.SourceContextAttribute;
                        });
                }

                _logger = loggingConfiguration.CreateLogger();
                loggingBuilder.AddSerilog(_logger, dispose: true);

                // Enable selflog to debug serilog configuration
                // SelfLog.Enable(Console.Error);
            });
```

**Comment by davidfowl** [comment, marked as answer]
What version of aspire are you using? 8.2?

**Reply by rcollette** [reply]
Yes.

**Reply by rcollette** [reply]
Nope, thought I was because I'm running the latest docker version but it is 8.1, and 8.2 [fixes it](https://github.com/dotnet/aspire/pull/5296) as pointed out on [stackoverflow](https://stackoverflow.com/a/79020605/107683).
